kbase-extension/static/kbase/js/util/jobLogViewer.js

```diff
@@ -802,11 +853,119 @@ define([
             }
         }
 
+        /**
+         * debugging function to display the current widget state
+         */
+        function renderWidgetState() {
+            if (developerMode) {
+                ui.setContent(
+                    'dev.widget-state',
+                    JSON.stringify(
+                        {
+                            fsm: fsm.getCurrentState().state,
+                            looping: looping,
+                            awaitingLog: awaitingLog,
+                            listeningForJob: listeningForJob,
+                        },
+                        null,
+                        1
+                    )
+                );
+            }
+        }
+
+        function renderJobStatusLines(jobState) {
+            const lines = Jobs.createJobStatusLines(jobState, showHistory);
+
+            return lines.map((line) =>
+                div(
+                    {
+                        class: `${cssBaseClass}__job_status_detail`,
+                    },
+                    line
+                )
+            );
+        }
+
+        function renderJobState(jobState) {
```

Comment:
This function renders the job state (surprisingly!)

If the job state is error, an errorDisplay element is added to show the error details.